ci/release/update-version.sh

```diff
@@ -37,60 +38,83 @@ function sed_runner() {
 
 # Centralized version file update
 echo "${NEXT_FULL_TAG}" > VERSION
+echo "${NEXT_FULL_TAG}" > RAPIDS_VERSION
 
-dependencies='libcuopt libcuopt-cu12 cuopt cuopt-cu12 cuopt-mps-parser'
-for FILE in conda/environments/*.yaml dependencies.yaml; do
-    for dependency in ${dependencies}; do
-        sed_runner "s/\(${dependency}==\)[0-9]\+\.[0-9]\+/\1${PY_NEXT_SHORT_TAG}/" "${FILE}"
-    done
+DEPENDENCIES=(
+  libcuopt
+  libcuopt-cu12
+  cuopt-mps-parser
+  cuopt
+  cuopt-cu12
+  cuopt-server
+  cuopt-server-cu12
+  cuopt-sh-client
+  cudf
+  cudf-cu12
+  cuvs
+  cuvs-cu12
+  librmm
+  librmm-cu12
+  libraft-headers
+  rmm
+  rmm-cu12
+  pylibraft
+  pylibraft-cu12
+  raft-dask
+  raft-dask-cu12
+  rapids-dask-dependency
+)
+
+for DEP in "${DEPENDENCIES[@]}"; do
+  for FILE in dependencies.yaml conda/environments/*.yaml python/*/pyproject.toml; do
+    sed_runner "s/\(${DEP}==\)[0-9]\+\.[0-9]\+/\1${NEXT_SHORT_TAG_PEP440}/" "${FILE}"
```

Comment:
```suggestion
    sed_runner "/-.* ${DEP}\(-cu[[:digit:]]\{2\}\)\{0,1\}\(\[.*\]\)\{0,1\}==/ s/==.*/==${NEXT_SHORT_TAG_PEP440}.*,>=0.0.0a0/g" "${FILE}"
```

Let's please use this pattern used throughout the rest of RAPIDS ([cudf example](https://github.com/rapidsai/cudf/blob/4a309b436685c605404fd82983673c1e94b7cd8c/ci/release/update-version.sh#L66)) which has the following nice properties:

* handles `-cu12` suffixes with an optional group, so no need to hard-code a bunch of `-cu12` entries in `DEPENDENCIES` (which would soon also need `-cu13` entries!)
* adds alpha specs to dependencies, so nightlies can be considered